src/Util.hs

```diff
@@ -188,3 +188,10 @@ checkTruthValueRangeAndSynTreeConf range synTreeConfig = do
   checkTruthValueRange range
   checkSynTreeConfig synTreeConfig
   pure ()
+
+vectorOfUniqueBy :: Int -> (a -> a -> Bool) -> Gen a -> Gen [a]
+vectorOfUniqueBy 0 _ _ = pure []
+vectorOfUniqueBy amount p gen = do
+  xs <- vectorOfUniqueBy (amount - 1) p gen
+  x <- gen `suchThat` \x' -> all (\y -> not (p x' y)) xs
+  pure $ xs ++ [x]
```

Comment:
Why not `x : xs`? That would be slightly more efficient, and the order in the list should not really matter.